XAES-256-GCM.go

```diff
@@ -28,6 +29,66 @@ const OverheadWithManualNonces = 16
 const Overhead = 40
 
 type xaes256gcm struct {
+	*xaes256gcmManual
+}
+
+// New returns a new XAES-256-GCM instance that automatically manages
+// generating/prepending/extracting nonces. As such, zero length nonces should
+// be passed in to [Seal] and [Open].
+func New(key []byte) (cipher.AEAD, error) {
+	x, err := newWithManualNonces(key)
+	if err != nil {
+		return nil, err
+	}
+
+	return xaes256gcm{x}, nil
+}
+
+func (xaes256gcm) NonceSize() int {
+	return 0
+}
+
+func (xaes256gcm) Overhead() int {
+	return Overhead
+}
+
+func (x xaes256gcm) Seal(dst, nonce, plaintext, additionalData []byte) []byte {
+	if len(nonce) != 0 {
+		panic("xaes256gcm: bad nonce length")
+	}
+
+	if total := len(dst) + len(plaintext) + Overhead; cap(dst) < total {
+		tmp := make([]byte, len(dst), total)
+		copy(tmp, dst)
+		dst = tmp
+	}
+
+	nonce = dst[len(dst) : len(dst)+NonceSize]
+	if _, err := rand.Read(nonce); err != nil {
```

Comment:
If the tmp path is not taken, and `dst = plaintext[:0]`, this will overwrite the plaintext.